src/pages/blog.astro

```diff
@@ -29,32 +28,9 @@ const posts = (await getCollection("blog")).sort(
         aria-labelledby="recent-posts"
         class="grid gap-6 sm:grid-cols-2 xl:grid-cols-3 items-start"
       >
-        {
-          posts.map((post) => (
-            <article class="bg-arg-white p-6 sm:p-8 rounded border-4 border-arg-muted shadow-md shadow-arg-muted/40 grid gap-4">
-              <div>
-                <FormattedDate
-                  date={post.data.date}
-                  classes="uppercase text-arg-gold"
-                />
-                <Heading tag={"h2"} size={"h4"}>
-                  <a
-                    class="font-bold focus:outline-none focus-visible:ring-4 ring-offset-2 ring-offset-arg-white ring-arg-gold"
-                    href={`/${post.slug}/`}
-                  >
-                    {post.data.title}
-                  </a>
-                </Heading>
-              </div>
-              <div class="bg-arg-muted h-[1px] w-full" />
-              <p>{post.data.description}</p>
-              <Link href={`/${post.slug}/`} type={"btn--gold"}>
-                Read Post
-              </Link>
-            </article>
-          ))
-        }
+        {posts.map((post) => <PostCard post={post} key={post.slug} />)}
       </section>
+      <SearchWidget />
```

Comment:
Should we lazy load this with a client directive like `client:idle` or whatever you recommend?

Reply:
I'd decided to make this widget an Astro component, so it'll just render as straight HTML (i.e., it's not a React component). It's basically a fancy redirect to the /search route with the React code that actually does all the searching.

Reply:
ooooo I see now! Nice 🙌